File: FlightETS-Java/app/src/main/java/com/example/flightets/SearchFlightActivity.java
package com.example.flightets;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.example.flightets.AuthApiService;


import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SearchFlightActivity extends AppCompatActivity {

    private EditText destinationInput, dateInput;
    private Button searchButton;
    private RecyclerView resultsRecyclerView;
    private TextView noResultsText;

    private FlightAdapter flightAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search_flight);

        destinationInput = findViewById(R.id.destinationInput);
        dateInput = findViewById(R.id.dateInput);
        searchButton = findViewById(R.id.searchButton);
        resultsRecyclerView = findViewById(R.id.resultsRecyclerView);
        noResultsText = findViewById(R.id.noResultsText);

        flightAdapter = new FlightAdapter(List.of());
        resultsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        resultsRecyclerView.setAdapter(flightAdapter);

        searchButton.setOnClickListener(view -> {
            String destination = destinationInput.getText().toString();
            String date = dateInput.getText().toString();

            if (destination.isEmpty() || date.isEmpty()) {
                Toast.makeText(this, "Veuillez remplir tous les champs", Toast.LENGTH_SHORT).show();
                return;
            }

            searchFlights(destination, date);
        });
    }

    private void searchFlights(String destination, String date) {
        SearchRequest request = new SearchRequest(destination, date);

        Call<List<Flight>> call = RetrofitClient.getInstance()
                .searchFlights(request);

        call.enqueue(new Callback<List<Flight>>() {
            @Override
            public void onResponse(Call<List<Flight>> call, Response<List<Flight>> response) {
                if (response.isSuccessful() && response.body() != null && !response.body().isEmpty()) {
                    flightAdapter = new FlightAdapter(response.body());
                    resultsRecyclerView.setAdapter(flightAdapter);
                    noResultsText.setVisibility(View.GONE);
                    resultsRecyclerView.setVisibility(View.VISIBLE);
                } else {
                    noResultsText.setVisibility(View.VISIBLE);
                    resultsRecyclerView.setVisibility(View.GONE);
                }
            }

            @Override
            public void onFailure(Call<List<Flight>> call, Throwable t) {
                Toast.makeText(SearchFlightActivity.this, "Erreur lors de la recherche", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: FlightETS/app/src/main/java/com/example/flightets/SearchFlightActivity.kt
package com.example.flightets

import android.app.DatePickerDialog
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class SearchFlightActivity : AppCompatActivity() {

    private lateinit var destinationInput: EditText
    private lateinit var dateInput: EditText
    private lateinit var searchButton: Button
    private lateinit var resultsRecyclerView: RecyclerView
    private lateinit var noResultsText: TextView

    private val flightAdapter = FlightAdapter(emptyList())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search_flight)

        destinationInput = findViewById(R.id.destinationInput)
        dateInput = findViewById(R.id.dateInput)
        searchButton = findViewById(R.id.searchButton)
        resultsRecyclerView = findViewById(R.id.recyclerView)
        noResultsText = findViewById(R.id.noResultsText)

        resultsRecyclerView.layoutManager = LinearLayoutManager(this)
        resultsRecyclerView.adapter = flightAdapter

        // Show DatePickerDialog when clicking dateInput
        dateInput.setOnClickListener {
            val calendar = Calendar.getInstance()
            val datePicker = DatePickerDialog(
                this,
                { _, year, month, dayOfMonth ->
                    val formattedDate = String.format("%04d-%02d-%02d", year, month + 1, dayOfMonth)
                    dateInput.setText(formattedDate)
                },
                calendar.get(Calendar.YEAR),
                calendar.get(Calendar.MONTH),
                calendar.get(Calendar.DAY_OF_MONTH)
            )
            datePicker.show()
        }

        searchButton.setOnClickListener {
            val destination = destinationInput.text.toString()
            val date = dateInput.text.toString()

            if (destination.isBlank() || date.isBlank()) {
                Toast.makeText(this, "Remplissez tous les champs", Toast.LENGTH_SHORT).show()
            } else {
                searchFlights(destination, date)
            }
        }
    }

    private fun searchFlights(destination: String, date: String) {
        val request = SearchRequest(destination, date)

        RetrofitClient.instance.searchFlights(request).enqueue(object : Callback<List<Flight>> {
            override fun onResponse(call: Call<List<Flight>>, response: Response<List<Flight>>) {
                if (response.isSuccessful && response.body() != null) {
                    val flights = response.body()!!
                    if (flights.isEmpty()) {
                        noResultsText.visibility = TextView.VISIBLE
                        flightAdapter.setFlights(emptyList())
                    } else {
                        noResultsText.visibility = TextView.GONE
                        flightAdapter.setFlights(flights)
                    }
                } else {
                    Toast.makeText(
                        this@SearchFlightActivity,
                        "Erreur lors de la recherche",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }

            override fun onFailure(call: Call<List<Flight>>, t: Throwable) {
                Toast.makeText(
                    this@SearchFlightActivity,
                    "Erreur: ${t.message}",
                    Toast.LENGTH_LONG
                ).show()
            }
        })
    }
}
